presto-docs/src/main/sphinx/admin/tuning.rst

```diff
@@ -5,35 +5,548 @@ Tuning Presto
+
+
+``optimizer.optimize-single-distinct``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Boolean``
+ * **Default value:** ``true``
+ * **Description:** Enables single distinct optimization. This optimization allows to perform applying distinct mask only once in cases where it's possible. This optimization will try to use single GROUP BY instead of multiple DISTINCT clauses. Enabling this optimization should speed up some specific selects but analyzing all queries to check if they qualify for this optimization may be a slight overhead.
+
+
+``optimizer.push-table-write-through-union``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Boolean``
+ * **Default value:** ``true``
+ * **Description:** Serves as default value for ``push_table_write_through_union`` session property. Parallelize writes when using UNION ALL in queries that write data. This allows to improve speed of writing output tables in UNION ALL clause by making use of the fact, that UNION ALL outputs do not require additional synchronization when collecting results. Enabling this optimization can improve UNION ALL speed when write speed is not yet saturated. However it's may slow down queries in already heavy loaded system.
+
+
+.. _tuning-pref-query:
+
+Query execution properties
+--------------------------
+
+
+``query.execution-policy``
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (``all-at-once`` or ``phased``)
+ * **Default value:** ``all-at-once``
+ * **Description:** Serves as default value for ``execution_policy`` session property. Setting this value to ``phased`` will allow query scheduler to split a single query execution between different time slots. This will allow to switch context more often and possibly stage the partially executed query in order to increase robustness. Average time of executing query may slightly increase after setting this to ``phased`` due to context switching and more complex scheduling algorithm but drop in variation of query execution time is expected.
+
+
+``query.initial-hash-partitions``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Integer``
+ * **Default value:** ``100``
+ * **Description:** Serves as default value for ``hash_partition_count`` session property. This value is used to determine how many nodes may share the same query when partitioning system is set to ``FIXED``. Manipulating this value will allow to distribute work between nodes properly. Value lower then number of presto nodes may lower the utilization of cluster in low traffic environment. Setting the number to to high value will cause assigning multiple partitions of same query to one node or ignoring the setting - in some configurations the value is internally capped at number of available worker nodes.
```

Comment:
What property would be used to set the partitioning system to ``FIXED`` ?

Which configurations cause the value to be capped at the number of available worker nodes`` ?

This value is used to determine how many nodes may share the same query when partitioning system is set to ``FIXED``. Manipulating this value will affect the distribution of work between nodes.  A value lower then the number of presto nodes may lower the utilization of the cluster in a low traffic environment. An excessively high value will cause multiple partitions of the same query to be assigned to a single node, or Presto may ignoring the setting.  In some configurations the value is internally capped at the number of available worker nodes.  This can also be specified on a per-query basis using the ``hash_partition_count`` session property.